src/runtime_src/core/edge/drm/zocl/zocl_kds.c

```diff
@@ -337,35 +385,54 @@ int zocl_command_ioctl(struct drm_zocl_dev *zdev, void *data,
 		goto out;
 	}
 	xcmd->cb.free = kds_free_command;
+	xcmd->cb.notify_host = notify_execbuf;
+	xcmd->execbuf = (u32 *)ecmd;
+	xcmd->gem_obj = gem_obj;
 
 	//print_ecmd_info(ecmd);
 
-	/* TODO: one ecmd to one xcmd now. Maybe we will need
-	 * one ecmd to multiple xcmds
-	 */
-	if (ecmd->opcode == ERT_CONFIGURE)
-		cfg_ecmd2xcmd(to_cfg_pkg(ecmd), xcmd);
-	else if (ecmd->opcode == ERT_START_CU)
+	switch(ecmd->opcode) {
+	case ERT_CONFIGURE:
+		xcmd->status = KDS_COMPLETED;
+		xcmd->cb.notify_host(xcmd, xcmd->status);
+		goto out1;
+	case ERT_START_CU:
 		start_krnl_ecmd2xcmd(to_start_krnl_pkg(ecmd), xcmd);
-	else if (ecmd->opcode == ERT_EXEC_WRITE)
-	{
+		break;
+	case ERT_EXEC_WRITE:
 		/* third argument in following function is representing number of
 		 * words to skip when configuring CU. This should be consistent
 		 * for both edge/DC, but due to performance and siome use cases,
 		 * this has veen decided that , DC flows skip 6 words whereas
 		 * edge flows doesnt skip any words.
 		 */
 		exec_write_ecmd2xcmd(to_start_krnl_pkg(ecmd), xcmd, 0);
-	}
-	else if (ecmd->opcode == ERT_START_FA)
+		break;
+	case ERT_START_FA:
 		start_fa_ecmd2xcmd(to_start_krnl_pkg(ecmd), xcmd);
-	xcmd->cb.notify_host = notify_execbuf;
-	xcmd->gem_obj = gem_obj;
+		break;
+	case ERT_START_COPYBO:
+		ret = copybo_ecmd2xcmd(zdev, filp, to_copybo_pkg(ecmd), xcmd);
+		if (ret)
+			goto out1;
+		goto out;
+	default:
+		DRM_ERROR("Unsupport command\n");
+		ret = -EINVAL;
+		goto out1;
+	}
 
 	/* Now, we could forget execbuf */
 	ret = kds_add_command(&zdev->kds, xcmd);
+	return ret;
 
+out1:
```

Comment:
literally, I don't "like" out1, out2, out etc. We'd better have a reasonable name and the next developer can follow what you mean and add his "out2" to a reasonable name. 

Given the scheduler is tied, we might fix this later.

Reply:
This is a good suggestion, let me enhance this later